weave/scorers/trust_scorer.py

```diff
@@ -0,0 +1,376 @@
+                    "WeaveFluencyScorerV1": 0.95,
+                    "WeaveCoherenceScorerV1": 0.9
+                }
+            }
+        )
+        ```
+
+    """
+
+    # Model configuration
+    device: str = Field(
+        default="cpu",
+        description="Device for model inference ('cpu', 'cuda', 'mps', 'auto')",
+        from_default=True,
+    )
+    context_relevance_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the context relevance model",
+        validate_default=True,
+    )
+    hallucination_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the hallucination model",
+        validate_default=True,
+    )
+    toxicity_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the toxicity model",
+        validate_default=True,
+    )
+    fluency_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the fluency model",
+        validate_default=True,
+    )
+    coherence_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the coherence model",
+        validate_default=True,
+    )
+    run_in_parallel: bool = Field(
+        default=True,
+        description="Whether to run scorers in parallel or sequentially, useful for debugging.",
+    )
+
+    # Define scorer categories
+    _critical_scorers: set[type[weave.Scorer]] = {
+        WeaveToxicityScorerV1,
+        WeaveHallucinationScorerV1,
+        WeaveContextRelevanceScorerV1,
+    }
+    _advisory_scorers: set[type[weave.Scorer]] = {
+        WeaveFluencyScorerV1,
+        WeaveCoherenceScorerV1,
+    }
+
+    # Private attributes
+    _loaded_scorers: dict[str, weave.Scorer] = PrivateAttr(default_factory=dict)
+    _emoji_pattern: re.Pattern = PrivateAttr(
+        default=re.compile(
+            "["
+            "\U0001f600-\U0001f64f"  # emoticons
+            "\U0001f300-\U0001f5ff"  # symbols & pictographs
+            "\U0001f680-\U0001f6ff"  # transport & map symbols
+            "\U0001f1e0-\U0001f1ff"  # flags (iOS)
+            "\U00002702-\U000027b0"  # dingbats
+            "\U000024c2-\U0001f251"
+            "]+",
+            flags=re.UNICODE,
+        )
+    )
+
+    def model_post_init(self, __context: Any) -> None:
+        """Initialize scorers after model validation."""
+        super().model_post_init(__context)
+        self._load_scorers()
+
+    def _load_scorers(self) -> None:
+        """Load all scorers with appropriate configurations."""
+        base_params: dict[str, Any] = {
+            "column_map": self.column_map,
+            "device": self.device,
+        }
+
+        # Load all scorers (both critical and advisory)
+        all_scorers = self._critical_scorers | self._advisory_scorers
+
+        for scorer_cls in all_scorers:
+            scorer_params = base_params.copy()
```

Comment:
Do you need this?  You are instantiating base_params fresh every time, so this is just an extra copy

Reply:
removed